### PR TITLE
Fix bootstrapper wrapper not being reset correctly

### LINKED ISSUE #2556: BootStrapper.reset() does not reset properly
## 🐛 Bug

Calling `BootStrapper.reset()` does not reset `self.metrics` properly.

### To Reproduce

```py
import torch
from torchmetrics.wrappers import BootStrapper
from torchmetrics.classification import MulticlassAccuracy

metric = BootStrapper(MulticlassAccuracy(num_classes=10))

for i in range(10):
    output = torch.randn((2000, 10))
    target = torch.randint(10, (2000,))
    # output = 0.5 * (target + output)
    metric.update(output, target)

print(metric.compute())
# {'mean': tensor(0.0990), 'std': tensor(0.0029)} <-- ok
print(metric.metrics[0].update_count)
# 10                                              <-- ok

metric.reset()

print(metric.compute())
# {'mean': tensor(0.0990), 'std': tensor(0.0029)} <-- ERROR, should be undefined after reset
print(metric.metrics[0].update_count)
# 10                                              <-- ERROR, should be 0 after reset

```

### Environment

- TorchMetrics version 1.4.0.post0
- Python version 3.11.9
- torch version 2.2.1+cu118

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2556

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2574.org.readthedocs.build/en/2574/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by stancld — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki mind checking:
```
Failed example:
    from torchmetrics.classification import MulticlassAccuracy
    from torchmetrics.wrappers import BootStrapper
    # creating metrics
    wrapped_metric = BootStrapper(MulticlassAccuracy(num_classes=3))
    # sample prediction and GT
    target = torch.tensor([0, 2, 0, 2, 0, 1, 0, 2])
    preds = torch.tensor([2, 1, 2, 0, 1, 2, 2, 2])
    # showing the metric results
    print(wrapped_metric(preds, target))  # this returns a dict with label info
Expected:
    {'mean': tensor(0.1476), 'std': tensor(0.0613)}
Got:
    {'mean': tensor(0.1333), 'std': tensor(0.1554)}

**Comment by SkafteNicki** [comment]
> @SkafteNicki mind checking:
> 
> ```
> Failed example:
>     from torchmetrics.classification import MulticlassAccuracy
>     from torchmetrics.wrappers import BootStrapper
>     # creating metrics
>     wrapped_metric = BootStrapper(MulticlassAccuracy(num_classes=3))
>     # sample prediction and GT
>     target = torch.tensor([0, 2, 0, 2, 0, 1, 0, 2])
>     preds = torch.tensor([2, 1, 2, 0, 1, 2, 2, 2])
>     # showing the metric results
>     print(wrapped_metric(preds, target))  # this returns a dict with label info
> Expected:
>     {'mean': tensor(0.1476), 'std': tensor(0.0613)}
> Got:
>     {'mean': tensor(0.1333), 'std': tensor(0.1554)}
> ```

I assume it is due to the randomness in the bootstrapper. I try setting a seed in the docs and update the values

**Comment by Borda** [comment]
I see, so we can seed the example...
